src/Smart.Design.Razor/TagHelpers/AcceptanceCheckbox/AcceptanceCheckboxHtmlGenerator.cs

```diff
@@ -0,0 +1,76 @@
+using System;
+using System.Globalization;
+using System.Threading;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Smart.Design.Razor.Resources;
+using Smart.Design.Razor.TagHelpers.Icon;
+
+namespace Smart.Design.Razor.TagHelpers.AcceptanceCheckbox;
+/// <inheritdoc/>
+public class AcceptanceCheckboxHtmlGenerator : IAcceptanceCheckboxHtmlGenerator
+{
+    private readonly IIconHtmlGenerator _iconHtmlGenerator;
+
+    public AcceptanceCheckboxHtmlGenerator(IIconHtmlGenerator iconHtmlGenerator)
+    {
+        _iconHtmlGenerator = iconHtmlGenerator;
+    }
+
+    /// <inheritdoc/>
+    public TagBuilder GenerateAcceptanceCheckbox(string label, AcceptanceCheckboxStyle style, string currentLanguage)
+    {
+        if (!string.IsNullOrEmpty(currentLanguage))
+        {
+            Thread.CurrentThread.CurrentUICulture = CultureInfo.CreateSpecificCulture(currentLanguage);
+        }
+        var iconAndLabel = style switch
+        {
+            AcceptanceCheckboxStyle.Danger => new Tuple<Image, string>(Image.Close, Translations.NotValidate),
+            AcceptanceCheckboxStyle.Warning => new Tuple<Image, string>(Image.Check, Translations.Validate),
+            _ => new Tuple<Image, string>(Image.Check, Translations.Validate)
+        };
```

Comment:
Juste une suggestion mais perso, j'aime bien nommé les champs de mes tuples et du coup ça me permet d'être moins verbeux dans l'instanciation de ma tuple comme ça
```suggestion
          (Image Icon, string Label) iconAndLabel = style switch
        {
            AcceptanceCheckboxStyle.Danger =>(Image.Close, Translations.NotValidate),
            AcceptanceCheckboxStyle.Warning => (Image.Check, Translations.Validate),
            _ => (Image.Check, Translations.Validate)
        };
```